e2e documentation updates

bsessions85 opened the pull request:
#### Changes proposed in this Pull Request

* Some updates to documentation related to the move of e2e tests

#### Testing instructions

* Ensure that the doc changes make sense and are complete

getdave committed:
Updates button wording to better convey action in context

Designer research indicates that Cancel is a better word to use in the context of “cancel this Import”.

See https://github.com/Automattic/wp-calypso/issues/30564

getdave committed:
Importers: Move all action buttons in to consistent space

Addresses https://github.com/Automattic/wp-calypso/issues/30564

spen committed:
Importers: Render close-button in importing pane if errored

spen committed:
Importers: Don't switch between uploading/importing panes prematurely

spen committed:
Importers: Truncate filename in upload pane

Commit:
e2e documentation updates

spen committed:
Importers: Prefix action button names

spen committed:
Importers: Update a translate context message, small code style tweaks

spen committed:
Importers: Always show continue button (disabled) throughout upload pane

Stojdza reviewed (commented):
Can you also change [this](https://github.com/Automattic/wp-calypso/pull/31468/files#diff-07b732fa7b580c0786ac5cc8384fc5ebR5) line to point to correct CircleCI run? 

Everything else looks good 👍

blowery committed:
Fix/migrate easy me styles (#31326)

* migrate me/account

* migrate me/account-password

* migrate me/application-passwords and me/application-password-item

* migrate me/account-close

* migrate me/two-step

* migrate me/security

* migrate confirm-cancel-domain

* migrate me/privacy

scruffian committed:
Move empty content css to the component (#31415)

* Move empty content css to the component

* Add comment

* Empty Content: use Button component

Co-authored-by: scruffian
Co-authored-by: jsnajdr

spen committed:
Fix e2e failure caused by importer button class change

marekhrabe committed:
New description for Medium importer (#31323)

lsinger committed:
Merge pull request #31331 from Automattic/add/sales-team-widget-to-plans-step

Signup: show sales team widget in plans step

bsessions85 committed:
Disabling Jetpack e2e test runs (#31488)

aldavigdis committed:
Extract wordpress externals 31351 (#31437)

* Extracting WordPress dependencies

- Moving most @WordPress external dependencies to the calypso-build package.
- babel-related dependencies stay in the main package.json file as they seem too interwoven and splitting them away requires more work than expected, so it requires a separate issue.

* the calypso-build package does not depend on any wordpress packages,
so we can remove those from package.json

* updating shrinkwrap

* removing an un-needed dependency

* reverting shrinkwrap changes

brandonpayton committed:
Expect auth cookie in support session (#31474)

* Previously, we did not permit the wordpress_logged_in cookie to be
  present with a support session during user bootstrap, but now, we
  need to permit and relay that cookie along with the session. This
  change updates user bootstrapping to always require the auth cookie
  and simplifies our `isLoggedIn` condition to rely solely on the
  cookie's presence.
* Fix unrelated linting errors in changed source file

andfinally committed:
Importers: Update WordPress file description. (#31482)

mattsherman committed:
Turn off autoCapitalize and autoCorrect on username fields (#31495)

deBhal committed:
enable show-tax feature flag on staging, horizon & wpcalypso (#31476)

taggon committed:
Signup: roll out improved onboarding test to all English users (#31384)

* Signup: roll out improved onboarding test to all English users

* Keep the default variation as is. Update the datestamp

* Update datestamp of improvedOnboarding test

* Update the test name in signup spec

jsnajdr committed:
Privacy Policy Banner: migrate CSS to webpack, useTranslate hook (#31479)

eliorivero committed:
Merge pull request #31465 from Automattic/update/jetpack-plans-table-sec-perf-emphasis

Plans: update Jetpack plans table with emphasis on security and performance

Aurorum committed:
Unread Notifications Accessibility

davemart-in committed:
Update related post demo images (#31518)

Before: https://cl.ly/0ad75b07be96
After: https://cl.ly/6dbaa142d1fb

ramonjd committed:
Signup: site type options are sometimes not translated (#31477)

* Moving the site type array into a function as a return value to give `translate` a chance to rerender new locales if necessary

* Updated jsdoc to remove old reference to `allSiteTypes`, which has now been deleted

Aurorum committed:
Whitespace 😬

tyxla committed:
Components: Update TimeSince to use Interval; remove obsolete Ticker (#31494)

* Components: Update TimeSince to use Interval

* Framework: Delete lib/ticker

tyxla committed:
Layout: AsyncLoad community translator (#31514)

Aurorum committed:
Use Long Content Fade (#31520)

sirreal committed:
Build: Remove webpack ignore props (#31525)

taggon committed:
Signup: Add TrainTracks integration to site vertical suggestions (#31466)

* Add TrainTracks integration to site vertical suggestions

* Update snapshot test

renovate[bot] committed:
fix(deps): update dependency node-sass-package-importer to v5.3.1 (#31515)

tyxla committed:
Components: Migrate PlansSkipButton style to Webpack (#31527)

blowery committed:
Migrate CSS for blocks/comments (#31502)

tyxla committed:
Jetpack Onboarding: Migrate JPO style to Webpack (#31528)

jsnajdr committed:
Use ESM version of react-virtualized and import modules individually (#31509)

They don't have `sideEffect: false`, so importing from the top-level package
is not efficient.

blowery committed:
Migrate CSS for my-sites/checkout

This is a quick and dirty migration, just porting the import statements from _components.scss to the corresponding root components. We could go further and break up the styles to match their components, but I'm not sure it's worth the additional complexity and risk with something as mission critical as the shopping cart.

blowery committed:
add missing subscription-text

bsessions85 committed:
Try Reporting e2e results to proper slack channel (#31501)

bsessions85 committed:
Update Chromedriver in e2e tests (#31489)

* Update Chromedriver in e2e tests

* Fix test broken by Chromedriver update

* Asyncify Customizer page

belcherj committed:
Update G Suite banner (#31469)

gititon committed:
Extract the file loader into a shard

blowery committed:
update import from test file

davewhitley committed:
Add explanation for rgba use

Co-Authored-By: torres126 <43215253+torres126@users.noreply.github.com>

Commit:
Remove CIrcleCI link

Commit:
Remove CIrcleCI link

Stojdza reviewed (approved):
LGMT 👍

blowery committed:
Merge pull request #31517 from torres126/fix/unread-accessibility

Colour Scheming: Unread Notifications Accessibility

jsnajdr committed:
Concierge calendar: add explicit import of moment-timezone (#31483)

Rather from importing moment.js from `i18n-calypso` and assuming it's `moment-timezone`
(it soon won't be), import it explicitly.

Stojdza committed:
Make sidebar selector more specific (#31497)

* Make sidebar selector more specific

* Dismiss component popover

keoshi committed:
Jetpack onboarding: add dark color scheme to onboarding flow (#31141)

* Add JP onboarding variables to color-schemes

* Add targetable className to Jetpack logo

* Add Jetpack onboarding styles

* Update spinner colors

* Update monthly/yearly billing selector

* Update get help link

* Adding more styles and packing everything neatly at the end of the file

* Fix PayPal logo visibility

* Add checkout styles

* Improve legibility on footer links

* Clean up plan tables

* Add thank you step styles

* Add styles for back button

* Add spaces inside parenthesis

* Add styles for progress bar that match the JP brand

* Update progress bar colors to deeper tones

* Apply onboading styles to layout element instead of body

* Update checkmarks/links colors in plans table

* Fix color for translation invite link

* Update site URL input focus styles

* Update TOS links color

* Apply segment control hover styles to focus as well

* Add card styles to authorize page error

* Move common button styles to the bottom

* Add selection styles

* Add missing styles for inputs and links

* Apply focus styles to all buttons equally

* Add `:not` pseudos to target onboarding flows coming from mobile apps

Thanks to @tyxla on https://github.com/Automattic/wp-calypso/pull/31454

ramonjd committed:
remove console log :) (#31548)

simison committed:
Switch package.json files to use tabs (#31543)

- Switched root and sub-folder package.json files to use tabs
   ```
    npx prettier \
      ./**/npm-shrinkwrap.json \
      ./**/package-lock.json \
      ./**/package.json \
      --write
   ```
- Removed editorconfig rule — left `*.yml` rule there
- Updated monorepo doc (indentation & typo)

keoshi committed:
Override styles for buttons that are disabled (#31549)

jsnajdr committed:
Checkout: fix ESLint warnings in TermsOfService

sirreal committed:
Jetpack blocks: Fix lints (#31544)

* Fix lints

```sh
npx eslint --ext .js --ext .jsx packages/jetpack-blocks/
```

```
…/packages/jetpack-blocks/src/blocks/contact-info/email/save.js
  10:47  error  Unnecessary escape character: \/  no-useless-escape
  10:53  error  Unnecessary escape character: \^  no-useless-escape
  10:56  error  Unnecessary escape character: \*  no-useless-escape
  10:72  error  Unnecessary escape character: \[  no-useless-escape

…/packages/jetpack-blocks/src/blocks/gif/edit.js
  46:55  error  Unnecessary escape character: \.  no-useless-escape

…/packages/jetpack-blocks/src/blocks/map/lookup/index.js
  168:9  error  Redundant double negation  no-extra-boolean-cast
```

* Fix lint

…/packages/jetpack-blocks/src/preset/setup/public-path.js
  9:2  error  Read-only global '__webpack_public_path__' should not be modified  no-global-assign

blowery committed:
Misc: Add CODEOWNERS to help automate reviews (#31551)

* Misc: Add a codeowners file to help automate finding reviewers

This is blank to start. Ideally, all assignments are of the form /path @team

* Add team-calypso interests

* update wording, add note about using teams

belcherj committed:
Allow purchase of extra seats for existing G Suite customers with mapped out domains (#31511)

sirreal committed:
CI: Prime early, wait less (#31329)

enejb committed:
Extracted the webpack and build into local webpack config (#31503)

Extract the webpack config object from calypso's webpack.config.js and merge it with the build script that is coming from jetpack-blocks package.

Co-Authored-By: enejb <enej.bajgoric@gmail.com>
Co-Authored-By: lezama <lezama@gmail.com>
Co-Authored-By: simison <mikael@ihminen.org>
Co-Authored-By: sirreal <jon.surrell@automattic.com>

ockham committed:
Monorepo: Move `bin/build-package.js` script to `calypso-build` package (#31542)

* Move bin/build-package.js
* Make build-package binary available
* Explicitly provide babel.config.js path

cbauerman committed:
Add QueryGSuiteUsers Component (#31444)

sgomes committed:
Prevent duplicate gridicons.

Gridicons were being bundled and loaded more than once.
This is because the gridicons module includes an `index.js` and an
`example.js`, beyond the individual icon files.
These files were being split out to a separate chunk but still loaded,
resulting in ~20KB of wasted JS.
The fix is to explicitly exclude `index.js` and `example.js`.

blowery committed:
Update client/components/async-gridicons/index.jsx

Missed escape.

Co-Authored-By: sgomes <sergiomdgomes@gmail.com>

sgomes committed:
Use global browserslist config for Babel. (#31561)

Babel had its own target configuration, which was a duplicate of the one
in package.json.

This removes the Babel-specific one, making Babel fall back to the one
in package.json.

blowery committed:
Merge pull request #31558 from Automattic/update/fix-gridicons-loading-multiple-times

Prevent duplicate gridicons.

shaunandrews committed:
Moving the error message to a notice, and adjust the styles of the input to match other parts of signup. (#31496)

blowery committed:
Merge pull request #31531 from Automattic/fix/migrate-checkout-css

Migrate CSS for my-sites/checkout

shaunandrews committed:
Increasing the contrast of the text color so its legible. Also removing the uppercase transform to make it easier to read. (#31280)

belcherj committed:
Fixes linting issue (#31564)

* Fixes linting issue

* Add block to remove unnecesary const declaration

* Remove accidentally added class

shaunandrews committed:
Signup: Remove radio's and button from `site-type` step (#31400)

Using clickable cards to replace radios and the button.

sirreal committed:
Fix minification (#31566)

shaunandrews committed:
This error isn't dismissable. (#31565)

blowery committed:
Build: Use gutenberg's browserlist (#31536)

Saves us one more bit of configuration.

bsessions85 committed:
Skipping broken import test (#31571)

fditrapani committed:
Remove address and phone number steps from signup (#31573)

deBhal committed:
Update/checkout paypal send tax location (#31484)

* setTax{Country,Postal}Code autovivification

* eslint

* Add defaultValue support to Input component

* Wire up paypal input to cartStore

* suppress autofocus lint warning

* fix tax location path

* Use separate prop for cart store update and revert to controlled component

* Fix pay button

deBhal committed:
Checkout Length Picker: Show generic message rather than tax amount (#31419)

shaunandrews committed:
Signup: Adding a list of common topics to the site-topic search. (#31405)

* Adding a list of common topics to the site-topic search.

* Update style.scss

* Making the suggestions and results positioned absolute over the site preview.

* Adding common topics component and associated styles.

Updating the common topics module with suggestions for each site type.

Adding a component update check to suggestion-search to test whether an incoming value is different from the current input value

Rounding off the bottom of the search suggestion drop down

* Updated verticals topic lists

Ensuring the topics component doesn't appear on the about step

Updating tests

* Removing _.isEmpty as we can add a simpler test, adding missing propType
Switching to PureComponent

* Updating snapshot

spen committed:
Extract SiteMockup Component for reuse

Copons committed:
Block Editor: Add an update image event on Media Modal changes (#31540)

Emit an updateImageBlocks event whenever an image has been updated or deleted in the Media Modal, so that the server can update accordingly all the blocks containing it.

jsnajdr committed:
Post Scheduler: reorganize moment-timezone usage (#31487)

ockham committed:
Add SEO descriptions to Gutenberg sidebar (#30033)

renovate[bot] committed:
chore(deps): update dependency eslint-plugin-jest to v22.4.1 (#31382)

renovate[bot] committed:
chore(deps): update dependency eslint to v5.15.3 (#31500)

renovate[bot] committed:
chore(deps): update jest monorepo to v24.5.0 (#31402)

renovate[bot] committed:
fix(deps): update dependency webpack-cli to v3.3.0 (#31504)

renovate[bot] committed:
fix(deps): update dependency webpack-dev-middleware to v3.6.1 (#30881)

jsnajdr committed:
Create library function to apply site offset and use it in Activity Log (#31537)

delputnam committed:
Show sale price/badge for domains on sale (#30821)

* Show sale price/badge for domains on sale

* Remove commented out code

* Better naming of CSS classes

* Tabs instead of spaces

* Show sale pricing for transfers.

* Fix spacing issues in scss file.

* Add flex-start to domain suggstion content in the featured suggestions.

* Check if sale coupon supports new domain purchases

* Rebase and update prop types to avoid error.

* Fixes from code review

* Don't show per year transfer pricing when initial query isn't set

sirreal committed:
Remove unused eslines config (#31590)

aidvu committed:
Domains: Remove privacy sub grouping from billing history

Privacy subs are gone. Don't group them anymore.

ockham committed:
Jest: Add per-package Jest config (#31594)

Use `projects` in the root Jest config, and per-package `jest.config.js` files. The rationale is similar as for per-package build scripts -- greater customizability. One concrete example is different transpilation (`.babelrc.js` files) per package, which will play a role in #31464.

Co-Authored-By: sirreal <jon.surrell@gmail.com>

sirreal committed:
Packages: Rely on npm lifecycle to prepare packages (#31597)

Per recent conversation, this seems to be the most sane way to build packages regularly

shaunandrews committed:
Moving the notice below the input to avoid pushing the input when the notice displays. (#31598)

shaunandrews committed:
Signup: Design tweaks to user account step (#31559)

* Design tweaks.

* Ditch the card styling for the social signup section.

* Remove the "Log in now." text from the bottom.

* Removing some JP overrides that are no longer needed. Styling the site card.

* Adjust the spacing for links below the form.

cbauerman committed:
Add New G Suite Cancellation Dialog (#30944)

blowery committed:
i18n-calypso: update version to 4.0.0-alpha.1 (#31601)

belcherj committed:
G Suite: Updates the ToS resolution flow (#31562)

ramonjd committed:
Removing absolute positioning for the form fieldset. (#31609)

Minor formatting.

ramonjd committed:
Codeowners: Adding Onboarding Dev team for /client/signup (#31576)

* Adding Onboarding Dev team as codeowners for /client/signup

* Adding Onboarding Dev team as codeowners for inline help and the checklist

* Add notifications for onboarding-dev for client state and components

cbauerman committed:
Correctly Get G Suite Survey Data From State (#31611)

ramonjd committed:
Adding checklist site logo ab test exception (#31578)

ramonjd committed:
Adding checklist site logo ab test exception (#31579)

ramonjd committed:
Onboarding: add `url` to the close welcome prompt tracking (#31582)

* Add context to the close tracks action, that is, we want to know which page the user has decided not to visit.

* added default props

jsnajdr committed:
i18n-calypso: remove moment-timezone and move the dep to parent project  (#31595)

* i18n-calypso: remove moment-timezone and move the dep to parent project

* i18n-calypso: move moment.js tests to separate file

* i18n-calypso: remove moment-timezone tests

sirreal committed:
Packages: Lift devDependencies (#31606)

* Lift  to root or remove devDependencies

* Move color-studio from package dep to root devDep

Not really a dependency of the built package

* Update deps

Aurorum committed:
Prevent site lock from disappearing (#31604)

Aurorum committed:
Credit Card, Billing & Account Recovery Text Contrast (#31539)

* Billing History: Text Variables

* Account Recovery Variables

* No results colour

taggon committed:
Fix: Typo in a step action function (#31552)

* Fix: Typo in a step action function

* Do not call isSiteTopicFulfilled when vertical already set

* Prevent excessive calls to isSiteTypeFulfilled

* Fix typo

taggon committed:
Signup: Show business plans first for non blog domain in the launch site flow (#31589)

tyxla committed:
Layout: AsyncLoad NpsSurvey (#31498)

simison committed:
Docs: update monorepo docs (#31585)

See https://github.com/Automattic/wp-calypso/pull/31606/ for more information.

sirreal committed:
Packages: Update package versions for release (#31618)

sirreal committed:
Format currency: Make package public (#31621)

tyxla committed:
CODEOWNERS: Add Jetpack connect (#31622)

brbrr committed:
Add data-e2e attribute to site types (#31620)

akirk committed:
Fix Typo on E-Mail Forwarding Page (#31627)

bsessions85 committed:
Re-enable import test after fix and update to not use constant (#31633)

retrofox committed:
simple-payments: increase the number of posts
This commit increases the number of posts (buttons) by request from 20 (default vaule) to 100.

belcherj committed:
Refactors GSuiteUserItem (#31613)

Aurorum committed:
Begin overrides

Aurorum committed:
Remove site overides

Aurorum committed:
Site Selector Overrides

Aurorum committed:
Remove Site Selector Overrides

retrofox committed:
Merge pull request #31640 from Automattic/improve/simple-payments-buttons-list

Simple Payments: increase the number of posts

gravityrail committed:
Clean up support for G Suite Business

Aurorum committed:
Try some formatting

gravityrail committed:
Virtualise gapps plan in Email component

Aurorum committed:
Format

gravityrail committed:
Make G Suite Basic the default

gravityrail committed:
Clean up how we fetch the plan

gravityrail committed:
Spaces -> tabs

gravityrail committed:
Detect gapps_unlimited when formatting cart

gravityrail committed:
Merge pull request #31650 from Automattic/try/clean-up-g-suite-business

Clean up support for G Suite Business

sirreal committed:
Packages: Fix ignore in calypso-color-schemes (#31628)

* Add npmignore to skip gitignore

* Update .npmignore

simison committed:
Add Gutenpack as codeowners for Jetpack blocks and Calypso build scripts (#31630)

simison committed:
Update monorepo package.jsons and documentation (#31623)

* Add directory key to monorepo packages

* spacing

* Update license SPDX from deprecated to new one in babel-plugin-i18n-calypso

* Update publishing docs

* Link Lerna

* Update docs/monorepo.md

Co-Authored-By: simison <mikael@ihminen.org>

* Update docs/monorepo.md

Co-Authored-By: simison <mikael@ihminen.org>

* Update docs/monorepo.md

Co-Authored-By: simison <mikael@ihminen.org>

* Update notes about logout/login

* unlogged -> un-authenticated

AnnaMag committed:
Jetpack Connect: Introduce NUX Site User Type step. (#31297)

* Include /site-owner route and controller. Add basic site schema.

* Include the content of the site ownershop screen.

Add proper options and adapt the copy.

* Rename the route, component and variables from site-owner to user-type.

* Redirect to the user type step from the site vertical one.

We hook up the new screen to the already merged NUX flow.

* Add feedback from review:

- remove obsolete prop
- remove the component from the wpcom NUX directory since it is not part of the wpcom signup

* Change the style, replacing radio buttons.

* Display user type only for JP > 9.2

include style definitions for the user type step. In the sequel, we will
remove possible code duplicates.

Aurorum committed:
Remove app promo overrides

Aurorum committed:
App Promo Override

Aurorum committed:
Switch styles around

belcherj committed:
Disallow domains with wpcomstaging in them from G Suite purchase (#31655)

mmtr committed:
Gutenframe: Preview posts at multiple screen sizes (#31610)

Use the Calypso's Preview component when previewing a post on Gutenframe so we can check how it will look like in different screen sizes.

In D25814-code we send a message to the parent frame that contains a `MessageChannel` port we listen to for knowing when the post has been autosaved and therefore can be previewed. The message of this port will containing the URL we can use for previewing the post and the post data needed for generating the "Search & Social" preview.

We also add a frame nonce to the preview URL since they are required by P2 posts.

ramonjd committed:
Onboarding: Add loading spinner for vertical search (#31614)

* Adding a spinner when the API is loading results

* adjusting spinner top position for mobile

ramonjd committed:
Checklist: ensure `calypso_checklist_task_display` is only fired when the task is displayed (#31583)

* By coupling the tracking event to the individual task, we can ensure that the tracking event `calypso_checklist_task_display` is only fired when the task in question is rendered. Adding `location` to the tracking props allows us to know where the task has been rendered, that is, on the stats page in the banner, in the inline help prompt, and so on.

* Remove the trackedTaskDisplays cache store since the continual tear down of the component wipes the cache anyway.

gravityrail committed:
Clean up logic for detecting invalid TLDs for G Suite

renovate[bot] committed:
chore(deps): update dependency eslint-plugin-import to v2.16.0 (#31663)

jsnajdr committed:
Migrate Activity Log CSS styles to webpack (#31629)

* Activity Log Filterbar: fix typo

* Activity Log Filterbar: use withLocalizeMoment HOC to format dates

* Activity Log Filterbar: migrate CSS to webpack

* Activity Log Tasklist: migrate CSS to webpack

* Activity Log Switch: migrate CSS to webpack

* Activity Log Example: migrate CSS to webpack

* Activity Log Banner: migrate CSS to webpack

- split CSS for intro, success and upgrade banners into individual files
- remove unused activity-log-banner__progress-bar style
- move activity-log__fader style to the right file

* Activity Log: separate styles for info buttons

A borderless gray button styles is used in confirm dialog and in error/success banners
for showing "More Info" and Happychat links.

Separate the styles into two distinct CSS classes. And remove unused CSS class names.

* Activity Log Confirm Dialog: migrate CSS to webpack

* Activity Log Threat Alerts: migrate CSS to webpack

Also a drive-by simplification of the alerts container: useTranslate hook, simpler code

* Activity Log: remove unused empty-day, first-sync and limit-notice styles

* Activity Log Rewind Toggle: migrate CSS to webpack (and fixup busy button)

* Activity Log: migrate CSS to webpack

* Activity Log: move confirm dialog styles to the correct file

* Activity Log Item: remove empty CSS rule (stylelint warning)

* Activity Log Item: migrate CSS to webpack

* Activity Log Confirm Dialog: move notice-content span to a better place

keoshi committed:
Onboarding: update site topic suggestions styling (#31619)

* Onboarding: add focus styles to site topic suggestions

* Onboarding: line up site topic suggestions with placeholder

* Remove empty line

arunsathiya committed:
Follow CSS namespace guidelines for search-preview (#31665)

AnnaMag committed:
Add /jetpack/connect/site-owner to the list of route ignore by JITMs. (#31447)

blowery committed:
Cart: Make some cart updates safer (#31644)

These can get triggered when a user adds a new credit card, but has no cart loaded.

Prevents an error from appearing on the add credit card page asking for a country, even though the country has been chosen.

sgomes committed:
Avoid re-rendering MomentProvider when mounting with current locale. (#31645)

There are some situations in which a new MomentProvider gets mounted,
e.g. when navigating to Stats. In this situation, it was being
rendered once with the hardcoded 'en' value, and once again with the
current locale if different from 'en'.

This change initializes MomentProvider to the currently active
moment locale ( moment.locale() ), effectively reusing the global
state rather than a hardcoded default.

Copons committed:
Masterbar: Prevent a full reload when clicking the Write button (#31626)

ockham committed:
Calypso Build Package: Add @automattic/calypso-color-schemes dep (#31563)

Rationale: The `calypso-build` package depends on the `calypso-color-schemes` package: https://github.com/Automattic/wp-calypso/blob/39202d0e87bb4bdda00cd643261e9aad6e3884dc/packages/calypso-build/postcss.config.js#L4

renovate[bot] committed:
fix(deps): update babel monorepo (#31572)

* fix(deps): update babel monorepo

* Babel config: specify corejs: 2 for preset-env

Needs to be specified explicitly in the latest Babel.
Also remove the `shippedProposal` flag, as there are currently no proposals shipped
in browsers that wouldn't already be in standard ES2018.

michaeldcain committed:
Add sub-header to site type step in Signup. (#31467)

blowery committed:
Merge branch 'update/layout-overrides' of https://github.com/torres126/wp-calypso into torres126-update/layout-overrides

* 'update/layout-overrides' of https://github.com/torres126/wp-calypso:
  Switch styles around
  App Promo Override
  Remove app promo overrides
  Format
  Try some formatting
  Remove Site Selector Overrides
  Site Selector Overrides
  Remove site overides
  Begin overrides

brbrr committed:
Add `data-e2e-slug` to user type cards (#31677)

blowery committed:
Merge pull request #31685 from Automattic/torres126-update/layout-overrides

Mirror of 31642: Torres126 update/layout overrides

jsnajdr committed:
People: migrate React component to ES6 and remove Jetpack version checks (#31453)

* PeopleSectionNav: fix ESLint warnings about variable declarations

* PeopleSectionNav: eliminate check for Jetpack version: all recent ones support team search

* PeopleInvites: eliminate Jetpack version check

* People: remove unneeded props and mixins and convert to ES6 class

The `People` class doesn't need the `peopleLog` prop and to watch the store:
only `PeopleNotices` need to do that. Allows us to remove the prop, the `observe`
mixin and to migrate the class to ES6.

Checking for Jetpack version is also no longer necessary, removing.

artpi committed:
Mailchimp: signup button (#31123)

This introduces a signup button to mailchimp box in sharing.
This will redirect to a mailchip signup form

* split button framework

* Button style changes based on feedback from @cburton4

* Removed Sass variable & reverted global primary button change based on @simison's feedback

* Add new link

* rebase fixes

sgomes committed:
Fix possible race conditions with localized-moment.

Following review by @jsnajdr in #31645.

bsessions85 committed:
Update gutenberg tests to use regular page preview (#31680)

* Update gutenberg tests to use regular page preview

* Add missing await

* Try with javascript click

* forgot post change

simison committed:
Remove Babel polyfill from Slideshow block (#31690)

belcherj committed:
Move email management components (#31653)

gravityrail committed:
Merge pull request #31662 from Automattic/fix/better-logic-for-excluding-gsuite-domain-suffix

Clean up logic for detecting invalid TLDs for G Suite

belcherj committed:
Remove domain only purchase Thank You screen (#31600)

* Remove domain only purchase Thank You screen
* Removes Domain only thank you
* Fix linting issue
* Update index.jsx

cbauerman committed:
Remove Confusing "Remove G Suite Purchase" Wait State (#31664)

* Use busy button for removing purchase

* Last Tweaks

* Tweaks to errors

* removed unused property

* Remove unused comments

mmtr committed:
Desktop App: Enable Gutenberg (#31580)

Enable the Gutenberg flags on the desktop environments, so the block editor becomes usable on the desktop app.

spen committed:
ImporterSignup: Handle loading notices in actions

spen committed:
ImporterSignup: Move submitStep & fetch mshot action logic to actions file

bluefuton committed:
Add Reader codeowners (#31660)

ramonjd committed:
Updating search field placeholder copy with instructional text. (#31656)

keoshi committed:
Related posts: update demo images (#31593)

Compressed images from: https://github.com/Automattic/jetpack/pull/11623

tyxla committed:
Jetpack Connect: Fix colophon alignment on user type step (#31710)

keoshi committed:
JP Onboarding: round the corners on the user type step (#31712)

* JP Onboarding: round the corners on the user type step

* Remove border bottom

* Use *-child instead of *-of-type

jsnajdr committed:
Jetpack Blocks: remove Babel Polyfill dependency from package.json (#31711)

The only usage got removed in #31690. Not needed any more in Jetpack Blocks.

renovate[bot] committed:
fix(deps): update react monorepo to v16.8.5 (#31695)

spen committed:
ImportSignupFlow: Fix whitespace issues around the error notice

blowery committed:
Build: Run distclean on stale installs (#31683)

* Install: Run a distclean on install to ensure consistency

ci cleans up node_modules at the root, but does not clean up nested node_modules in packages. This fixes that to also clean up nested node_modules, and run a normal clean.

* fix up processing spawnSync result and exit

scruffian committed:
Merge pull request #31334 from torres126/remove/google-plus-share-button

Share Button Component: Remove Google+

jsnajdr committed:
ProfileGravatar: convert to functional component, remove localize HOC (#31725)

renovate[bot] committed:
chore(deps): update dependency sinon to v7.3.0 (#30803)

ramonjd committed:
Vertical search: ensure results float over site preview (#31661)

* #31609 was a hotfix to ensure that Jetpack connect site vertical results weren't cut off. This PR reinstates the position: absolute so that the fieldset floats over the site preview, but also sets the layout overflow to visible on Jetpack to avoid cutting off the results list.

* Overwriting positioning on the form-fieldset to avoid overflow issues on smaller device heights. Props @keoshi

jeffersonrabb committed:
Slideshow block: Fix two Javascript Errors (#31720)

* Abort announcement if current slide is undefined. Fixes #31247

* Check for existence of swiper instance and element before resizing Slideshow block. Fixes #30876.

shaunandrews committed:
Signup: Update site-title singular step (#31641)

* Adjust the design of the single-input variant to match the site-topic step.

* Trying (and failing) to add a subHeaderText to the site-information-title step.

* Typo. This should have been 12px from the start.

* Was missing the fallback, which is apparently needed to get the subhead to display.

* Updating the title and subhead to be more descriptive about the site title input.

* Decided against updating these titles in this PR.

* Trying to appease the tests.

* Another random stab in the dark to fix the tests.

* Update snapshot.

* Missed a string.

creativecoder committed:
Enables engine6 imports in wpcalypso environment (#31730)

shaunandrews committed:
Updating the card to be more inline with the rest of signup, and removing some (I think, unused) translates that caused a horiz scrollbar. (#31652)

arunsathiya committed:
Lint fix: twitter-preview: Follow CSS namespace guidelines (#31666)

* Follow CSS namespace guidelines for twitter-preview

* Update classnames inside template strings to follow CSS namespace guidelines

AnnaMag committed:
Improve the copy of the NUX header, in alignment with the wpcom one. (#31706)

dbspringer committed:
Add "Hide ad on mobile views" toggle (#31608)

Placing control outside placeholder

bsessions85 committed:
Update/cleanup e2e lint (#31732)

* Remove linting exclusions

* next line

* Update to match linting rules from wp-calypso as much as possible

* bad reference

jblz committed:
Import in Signup: Split step dependencies so tracks event records (#31731)

* Import in Signup: Split step dependencies so tracks event records
* Record that a preview was fetched vs including a blob resource ID in event

bluefuton committed:
Recommended plugins: add data layer (#31135)

#### Changes proposed in this Pull Request

Works with the new `/sites/<slug>/plugins/recommended` API endpoint introduced in D24342-code by @skjaini.

<img width="1062" alt="Screen Shot 2019-03-13 at 12 59 48 PM" src="https://user-images.githubusercontent.com/4044428/54306836-e835f180-458f-11e9-8707-6a2b2922fcb3.png">

This adds a feature flag that replaces the Featured section with a dynamically curated Recommended section for the plugins page (/plugins/<slug>). Please note that the current implementation only supports Jetpack connected sites. This feature flag is enabled by default for wpcalypso and development environments.

This PR introduces:
- [x] a tested data layer implementation for the recommended plugins network request
- [x] a tested Redux state implementation (actions/reducer/selector)
- [x] a query component that dispatches a network request action upon mounting
- [x] data layer to do the API fetch

dbspringer committed:
Removing header/footer copy per @thomasguillot (#31739)

creativecoder committed:
Import: Use Modern Business theme for sites created while importing in signup (#31734)

jonathansadowski committed:
Import: remove site title from descriptive header (#31740)

* Import: remove site title from descriptive header

* Remove unused strong element

mmtr committed:
Gutenframe: Display success message after trashing a post - Take 2 (#31704)

Move the logic of trashing a post in Guteframe to Calypso. It reuses the available trashPost action that, apart from moving a post to the trash, displays a success message when the operation is done with the possibility to undo it.

ramonjd committed:
Signup: Update redirect url to go to checklist after site creation (#31744)

* Update redirect url to go to checklist
Remove double negation as per eslint's persuasive nudges

* Update method name so that it does what it says it does. We're not showing the preview anymore, but the checklist.

niranjan-uma-shankar committed:
Redirect to launch flow from Settings and Checklist (#31709)

* Redirect the Launch Site button in Settings page and Checklist page to the launch flow.

jsnajdr committed:
Domains: remove unused .icann-verification styles

The stylesheet was duplicated/migrated to `email-verification` in #19577 but never removed.

jsnajdr committed:
Domains: migrate .email-verification CSS styles to webpack

Also remove global override for `.notice__text` (added in #15880)

jsnajdr committed:
Domains: shorten the import path of IcannVerificationCard component

sirreal committed:
Jetpack blocks: Major version bump (#31748)

Copons committed:
Block Editor: use Calypso Media Modal in Classic Blocks (#31596)

* Add a new Media Modal instance to be used in Classic blocks.
* Make Calypso listen to `classicBlockOpenMediaModal` messages and respond with `insertClassicBlockMedia` ones.

arunsathiya committed:
Break SEO stylesheet into individual component stylesheets, useTranslate hook for SearchPreview component (#31733)

* useTranslate hook for SearchPreview component

* Break SEO stylesheet into individual component stylesheets

sgomes committed:
Memoize the last active discount. (#31678)

* Memoize the last active discount.

Lack of memoization in getActiveDiscount was causing the SiteNotice
component to re-render needlessly at a high rate.

This commit adds simple last value memoization in the getActiveDiscount
selector so that the previous value is reused if nothing's changed.

* Changes to getActiveDiscount memoization after review.

sgomes committed:
Fix possible race conditions with localized-moment. (#31691)

Following review by @jsnajdr in #31645.

arunsathiya committed:
Fix broken Twitter preview styling (#31753)

* Fix broken Twitter preview styling

* Update class name for component

shaunandrews committed:
Forcing 0 margins to avoid a spacing bug in Safari (#31735)

Stojdza committed:
Add NODE_CONFIG_ENV instead NODE_ENV (#31632)

* Add NODE_CONFIG_ENV instead NODE_ENV

* update package-lock

* Update warnings

* Update run-wrapper.sh

* Update circle-ci-branch-update.sh

* Update .circleci/config.yml

* Revert "Update run-wrapper.sh"

* Revert "Revert "Update run-wrapper.sh""

This reverts commit e308b10

* Update docs

scruffian committed:
Change the way we handle positioning when a guided tour will overlap the right side of the screen

sirreal committed:
Jetpack blocks: Remove Jetpack blocks artifact fetching (#31760)

This was used for an automated synchronization process that is disabled
and will not be restored.

Remove the fetch code and the integration test.

sirreal committed:
Jetpack blocks: Remove CircleCI build (#31757)

No longer in use.

sirreal committed:
SDK: Failed builds should exit with error status (#30019)

Stojdza committed:
[e2e] Upgrade config (#31764)

* Upgrade config

* update package-lock.json

aidvu committed:
Domains: Search not showing map/transfer option

We didn't limit the scope of a prop, so if it got stored
into local storage, it would still be available outside
of NUX, thus hiding the option from regular domain search.

cbauerman committed:
Fix Gsuite TOS Link (#31738)

cbauerman committed:
Switch Email and Domain Management Email Paths (#31703)

ramonjd committed:
Inline Popover: Fix/close onboarding welcome prompt (#31745)

* Ensuring the welcome prompt goes away whenever the popover is given the instruction to hide.

* added reducer tests

southp committed:
Add: the data layer handlers & actions for interacting with `/verticals` endpoint (#31714)

* Add data-layer handlers and the actions for interacting with
`/verticals` v2 API.

* Move the verticals data actions into its own `signup/verticals` folder,
since it doesn't really belong to any signup steps.

* Use `convertToCamelCase()` as the `fromApi()` function for `/verticals`
data-layer handler.

* Update the path for importing `setVerticals`

* Add the test suite for signup/verticals/actions

* Add the test suite for /signup/verticals data-layer handlers.

southp committed:
Add: verticals substate tree (#31716)

* Add the new `verticals` reducer

* Add `getVerticals` selector.

* Add a test suite for `signup/verticals/selectors`

* Add a test suite for the signup/verticals/reducer

southp committed:
Add: <QueryVerticals/> component (#31718)

* Implement <QueryVerticals/> component

* query-signup-verticals => query-verticals

* Make use of `componentDidUpdate()` life cycle method to make sure
`QueryVerticals` component is fetching per updates.

* Give searchTerm an empty string as the default

* Add a test suite for QueryVerticals component

southp committed:
Wiring the new `verticals` state tree to the related components (#31746)

* Plumb the underlying `signup.verticals` state tree onto
`SiteVerticalsSuggestionSearch` component

* Draft: making the site vertical data a derived data from selector,
rather than a vanilla state tree.

* Add a `isNavigating` parameter to the `onChange()` callback of the
component `SuggestionSearch`

* Store the candidate vertical list in a state so that a user can navigate
through it without updating the list, since the props `verticals` is
changing depending on the `searchValue` props.

* Derive `isVerticalSearchPending` from `verticals` props.

* Tweak the definition of isVerticalSearchPending.

* Move some selectors down and add a TODO comment to note down the future
polishing plan.

* Update the test suite of site-vertical/selectors

* Update the `about` step to accommodate the related changes of
<SiteVerticalsSuggestionSearch/>

* Fix the <SiteVerticalsSuggestionSearch /> component tests and remove
unused props.

* More TODOs

* Add the missing constants.js

* Remove the unused `lastUpdated`

* Use `getSiteVerticalName()` for getting the site vertical name.

* Fix a type

jsnajdr committed:
UpgradeNudgeExpanded: migrate CSS to webpack (#31758)

* UpgradeNudgeExpanded: remove Media/VideoPress styling

It was used in an A/B test that didn't succeed (#8403).

* UpgradeNudgeExpanded: migrate CSS to webpack

jsnajdr committed:
Domains: migrate ICANN/email verification CSS to webpack (#31751)

* Domains: remove unused .icann-verification styles

The stylesheet was duplicated/migrated to `email-verification` in #19577 but never removed.

* Domains: migrate .email-verification CSS styles to webpack

Also remove global override for `.notice__text` (added in #15880)

* Domains: shorten the import path of IcannVerificationCard component

brbrr committed:
E2E tests: Fix Jetpack connect (and others) tests  (#31668)

* Move e2e-related ingores to e2e folder

* Some code clean-up

* Update jetpack connect tests to reflect recent onboarding changes

* Add UserType JPO step

* More fixes here and there

* Dismiss a stats popup for invite test

sirreal committed:
Make SDK Gutenberg generic (#31723)

The SDK Gutenberg tool had become overly specific to build the Jetpack
Gutenberg extensions. Said extensions no longer depend on the SDK tool,
instead relying on their own specific build tool.

Remove the changes that were geared towards the Jetpack builds.

keoshi committed:
My Plan: reorder plan feature cards to highlight focus on security and performance (#31749)

* Tighted up current plan header

* Add class to plan header if is free plan

* Add conditional styles for free plan

* Reorder feature cards

* Fix plan icon vertical spacing in paid plans

* Bump the minimum required screen size to show the button on the right

* Hide plan icon on small screens

tyxla committed:
Components: Migrate HrWithText CSS to Webpack (#31794)

* Components: Migrate HrWithText CSS to Webpack

* Remove prettier pragma

sirreal committed:
Framework: Make JSON pretty too (#31554)

* Prettier all json
* Add some json to eslintignore

tyxla committed:
Blocks: Migrate EditGravatar CSS to Webpack (#31795)

* Blocks: Migrate EditGravatar CSS to Webpack

* Remove prettier pragma

keoshi committed:
Jetpack Onboarding: add accent color, clean up styles (#31625)

* Onboarding: add overriding color var for Jetpack flow

* Replace JP color for var

* Remove unnecessary overriding rules

* Move color overrides to the top, add more

* Delete unused styles

* Update color variables name

* Add overrides for checkout page header

sirreal committed:
Jetpack blocks: Remove 💥 (#31759)

* Remove jetpack-blocks package
* Remove babel override
* Remove from CODEOWNERS
* Remove now-obsolete mapbox-gl dep
* Remove unused copy-webpack-plugin
* Remove Jetpack blocks mention in client/gutenberg/extensions/README.md

tyxla committed:
Layout: Exclude atomic sites from JP logic (#31797)

Copons committed:
Block Editor: Update the featured image preview after editing the image file (#31761)

jsnajdr committed:
Domains: remove privacy-protection-card styles (#31796)

scruffian committed:
Remove the rect.width parameter when xAboveBelow is called, as its no longer used

matthusby committed:
Blogger Plan: Allow more then 1 `.blog` domain on the blogger plan. (#30999)

scruffian committed:
Merge pull request #31763 from Automattic/update/guided-tours-position

Change the way we handle positioning for guided tours

michaeldcain committed:
Post/Page List: add section headers with 'add new' buttons (#31783)

* Add section header with 'add' button to PostTypeList.

* Add section header with 'add' button to page list.

Aurorum committed:
WooCommerce Signup: Add Responsiveness (#31049)

* WooCommerce Signup: Add Responsiveness

* Comply Breakpoint with Guidelines

* Prevent the link and button looking too squashed

* Comply with guidelines by building for mobile

* Whitespace

cphilleo committed:
Prevent Jetpack sites displaying WordAds Instant Activation Toggle (#31743)

* Prevent Jetpack sites from showing WordAds instant activation toggle

* Use the correct plan name in the instant activation toggle

niranjan-uma-shankar committed:
Fix/broken css signup pages (#31803)

* dops is excluded but gravatar page breaks due to this

* Adjusting the padding and removing the extra background color.

* Making sure any links in the header are visible on the primary background.

kwight committed:
Block Editor: Add default routes for block editor paths. (#31788)

bsessions85 committed:
[e2e] Use active tests file as source for e2e a/b tests (#31809)

* Use abtests from Calypso

* cleanup a bit

* using map instead

* Handling test overrides

* needed a null check

* use wrapper in async base container

* Use abtests from Calypso

* cleanup a bit

* using map instead

* Handling test overrides

* needed a null check

* use wrapper in async base container

shaunandrews committed:
SCSS: Add an elevation mixin (#31470)

* Adding a mixin to let us add Material elevation with box-shadows.

* Update assets/stylesheets/shared/mixins/_elevation.scss

Co-Authored-By: shaunandrews <shaun@automattic.com>

* Adding a new --color-black variable. Using the hex value for both as Muriel doesn't define one.

* Using the new color variable for the shadows.

* Swapping out the box-shadow on the site-type step in signup to use the new mixin.

Commit:
Merge branch 'update/e2e-docs' of github.com:Automattic/wp-calypso into update/e2e-docs
